apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -227,38 +233,51 @@ external_binary_to_mess_id(BExtMessID) when is_binary(BExtMessID) ->
 %% -----------------------------------------------------------------------
 %% XML
 
+%% @doc Adds all arcid elements (for backward compatibility)
```

Comment:
I've met people very sensitive about sending "unused" data and this would apply to extra `stanza-id` element in existing apps, that do not use it yet. I still think that both elements should be configurable for now, with `archived` being deprecated (with clear warning in release notes and on server startup).